source/Calamari.Aws/Deployment/Conventions/UploadAwsS3Convention.cs

```diff
@@ -236,22 +234,36 @@ public S3UploadResult UploadUsingPackage(Func<AmazonS3Client> clientFactory, Run
         /// <param name="bucketKey"></param>
         /// <param name="properties"></param>
         /// <returns>PutObjectRequest with all information including metadata and tags from provided properties</returns>
-        private PutObjectRequest CreateRequest(string path, string bucketKey, S3TargetPropertiesBase properties)
+        private PutObjectRequest CreateRequest(string path, Func<string> bucketKey, S3TargetPropertiesBase properties)
         {
             Guard.NotNullOrWhiteSpace(path, "The given path may not be null");
             Guard.NotNullOrWhiteSpace(bucket, "The provided bucket key may not be null");
             Guard.NotNull(properties, "Target properties may not be null");
 
             return new PutObjectRequest
-                {
-                    BucketName = bucket?.Trim(),
-                    FilePath = path,
-                    Key = bucketKey?.Trim(),
-                    StorageClass = S3StorageClass.FindValue(properties.StorageClass?.Trim()),
-                    CannedACL = S3CannedACL.FindValue(properties.CannedAcl?.Trim())
+            {
+                FilePath = path,
+                BucketName = bucket?.Trim(),
+                Key = bucketKey()?.Trim(),
+                StorageClass = S3StorageClass.FindValue(properties.StorageClass?.Trim()),
+                CannedACL = S3CannedACL.FindValue(properties.CannedAcl?.Trim())
+            }
+            .WithMetadata(properties)
+            .WithTags(properties)
+            .WithMd5Digest(fileSystem);
+        }
+
+        private static Func<string> GetBucketKey(ICalamariFileSystem fileSystem, string filePath, IHaveBucketKeyBehaviour behaviour)
```

Comment:
Yea, good having this method, I was confused by why there were differences when calling CreateRequest previously, with the enum this helps